app/auth_helpers.py

```diff
@@ -26,40 +27,64 @@ def get_google_credentials(impersonate_email: str):
     credentials = service_account.Credentials.from_service_account_file(KEY_PATH,scopes=SCOPES).with_subject(impersonate_email)
     return credentials
 
-def authorize_admin(request: Request, token: HTTPAuthorizationCredentials = Depends(security_scheme)):
-    payload = authorize(request, token)
+def authorize_admin(request: Request):
+    payload = authorize(request)
     
     if payload.role != "admin":
         raise HTTPException(403, 'Insufficient privileges to access this resource')
     return payload
 
-def authorize(request: Request, token: HTTPAuthorizationCredentials = Depends(security_scheme)):
+def authorize(request: Request):
     '''
     Takes a request and goes through all the steps of authenticating it.
     On valid request, function will return the payload.
     If token used in the request is invalid, a number of errors can be raised
     and returned.
     '''
+    access_token = request.cookies.get("access_token")
+    if not access_token:
+        raise HTTPException(
+            401, 'Access token is not present')
+
     payload = AccessTokenPayload.parse_obj(decode_token(
-        token.credentials, request.app.config))
+        access_token, request.app.config))
+
     if not payload.access_token:
         raise HTTPException(
             401, 'The token is not an access token. Is this a refresh token?')
     return payload
 
-def optional_authentication(request: Request, token: HTTPAuthorizationCredentials = Depends(optional_security)):
+def optional_authentication(request: Request):
     '''
     Allows login to be optional, and if token is provided parse it using authorize. This gives the possibility for endpoints 
     to change behavior based on the user being logged in or not
     '''
-    if token:
-        return authorize(request, token)
-    return None
+    access_token = request.cookies.get("access_token")
+    if access_token:
+        try:
+            # Don't use decode_token function as it raises an exception which we
+            # do not want on optional authentication
+            payload = AccessTokenPayload.parse_obj(decode(access_token, request.app.config.SECRET_KEY, algorithms=['HS256']))
+            return payload.access_token and payload or None
+        except:
+            return None
+
+    else:
+        return None
 
 def role_required(accessToken: AccessTokenPayload, role: str):
     if accessToken.role != role:
         raise HTTPException(403, 'No privileges to access this resource')
 
+def set_auth_cookies(response: Response, access_token: str, refresh_token: str):
+    secure = os.environ.get("API_ENV") == "production" and True or False
```

Comment:
The purpose is to only enable the secure flag on production. This is because I believe it might cause issues to always have it to True, as under local development we are only using `http` and not `https`

Reply:
`secure=os.environ("API_ENV") == "production"` is true in prod and false otherwise so do we need that last part?

Reply:
Yes of course hahaha, I just realised 🤦